package/src/web_voice_processor.ts

```diff
@@ -9,157 +9,60 @@
-      downsamplingWorker,
-    ];
-  }
-
-  private async _setupAudio(): Promise<any> {
-    this._node = this._audioContext.createScriptProcessor(4096, 1, 1);
-    this._node.onaudioprocess = (event: AudioProcessingEvent): void => {
-      if (this._isRecording) {
-        this._downsamplingWorker.postMessage({
-          command: 'process',
-          inputFrame: event.inputBuffer.getChannelData(0),
-        });
-      }
-    };
-
-    this._audioSource.connect(this._node);
-    this._node.connect(this._audioContext.destination);
-
-    this._downsamplingWorker.onmessage = (
-      event: MessageEvent<DownsamplingWorkerResponse>,
-    ): void => {
-      switch (event.data.command) {
-        case 'output': {
-          for (const engine of this._engines) {
-            engine.postMessage({
-              command: 'process',
-              inputFrame: event.data.outputFrame,
-            });
-          }
-          break;
-        }
-        case 'audio_dump_complete': {
-          this._audioDumpResolve(event.data.blob);
-          this._audioDumpPromise = null;
-          this._audioDumpResolve = null;
-          this._audioDumpReject = null;
-          break;
-        }
-        default: {
-          console.warn(`Received unexpected command: ${event.data.command}`);
-          break;
-        }
-      }
-    };
+  private constructor(options: WebVoiceProcessorOptions) {
+    this._engines = new Set();
+    this._options = options;
+    this._vuMeterCallback = options.vuMeterCallback;
   }
 
   /**
-   * Acquires the microphone audio stream (incl. asking permission),
-   * and continuously forwards the downsampled audio to speech recognition worker engines.
+   * Gets the WebVoiceProcessor singleton instance.
    *
-   * @param options Startup options including whether to immediately begin
-   * processing, and the set of voice processing engines
-   * @return the promise from mediaDevices.getUserMedia()
+   * @param options Startup options.
+   * @return WebVoiceProcessor singleton.
    */
-
-  public static async init(
-    options: WebVoiceProcessorOptions,
+  public static async instance(
+    options: WebVoiceProcessorOptions = {},
   ): Promise<WebVoiceProcessor> {
-    const [microphoneStream, audioContext, audioSource, downsamplingWorker] = await this._initMic(options);
-
-    return new WebVoiceProcessor(
-      microphoneStream,
-      audioContext,
-      audioSource,
-      downsamplingWorker,
-      options,
-    );
-  }
-
-  private constructor(
-    inputMediaStream: MediaStream,
-    audioContext: AudioContext,
-    audioSource: MediaStreamAudioSourceNode,
-    downsamplingWorker: DownsamplingWorker,
-    options: WebVoiceProcessorOptions,
-  ) {
-    this._options = options;
-
-    if (options.engines === undefined) {
-      this._engines = [];
+    if (!this._instance) {
+      this._instance = new WebVoiceProcessor(options);
     } else {
-      this._engines = options.engines;
+      this._instance._options = options;
```

Comment:
if you change the options while the wvp is running, does anything get messed up?

Reply:
nope but the settings wont change until stop/start has been called

Reply:
Ah okay, might be good to document that then